### DISCUSSION TITLE
使用popover过程中遇到了点小麻烦

### DISCUSSION THREAD

**Discussion by ShangChien** [tusen-ai/naive-ui#3041] [Q&A] [answered]
![Inked152331_LI](https://user-images.githubusercontent.com/48350636/171580734-9c9a78b8-a321-4eb6-a217-29baef3f8e57.jpg)

在用`n-grid`展示`n-card`时，点击相应的`card`，`popover`弹出具体的内容描述。但是当`popover`弹出的内容超出浏览器的视图时，会激活浏览器原生的视窗滚轮（`n-layout`里禁用了原生的滚轮），这样就很不美观。

很自然想到把`popover`的位置设为`top`，但是内容描述好像不会跳出父元素`card`的框框，反而会挡住图片本身：
![152442](https://user-images.githubusercontent.com/48350636/171580170-ae341825-3f32-4ab9-b149-0d31c3cb8561.png)

后面仔细看了下官网naive UI的popover展示，当弹出内容碰到视图边框会自动调整。这是有什么细节需要注意吗？
（我使用naive UI是v2.29.0）

**Comment by 07akioni** [comment, marked as answer]
正常来说是会避让的，我也不太清楚在你的环境里是出现了啥问题。

浏览器有没有缩放？

最好能给个复现，不然很难确定发生了啥。

**Reply by 07akioni** [reply]
<img width="1448" alt="image" src="https://user-images.githubusercontent.com/18677354/171658939-81c93df1-017c-4470-91c6-98e83103ca90.png">

这个属性控制自动调整，但是默认是 true 的

**Reply by ShangChien** [reply]
简单回复一下，本来想写个在线的代码复现，但是简单的playground不能装naive_ui， 初学只会<script setup>，但是StackBlitz用起来总是报defineProps未定义。所以在线的代码复现卒。。。

不过，我自己又捣鼓了一会，似乎知道了出问题的原因。
- popover的内容位置不对，是父元素card的position和插槽内容的position用错了，顺便深入学了下position的五个参数用法。
- 原生滚动条的出现，可以通过设置`popover`的 `:to='false'`解决。不确信具体原因，但是在弹出内容过程中查看html元素变化，发现弹出元素是body的直接子元素，和我的layout 框架同级。而且dom元素不会消失。虽然浏览器中不显示，但似乎还是会计算高和宽，所以应该是这样挤出了原生浏览器的滚动条。(但是官网popover弹出的，也是body直接子元素。。。所以上面脑洞分析可能是胡说八道)

总结，我太菜了，还要继续认真学。

**Reply by 07akioni** [reply]
弹出在 body 的原因是因为不会被用户的 overflow 样式影响

**Reply by 07akioni** [reply]
to=false，一般是比较特殊的时候有需要的话再用
